tm2/pkg/sdk/context.go

```diff
@@ -171,10 +171,14 @@ func (c Context) Value(key interface{}) interface{} {
 // ----------------------------------------------------------------------------
 
 // Store fetches a Store from the MultiStore, but wrapped for gas calculation.
-func (c Context) Store(key store.StoreKey) store.Store {
+func (c Context) GasStore(key store.StoreKey) store.Store {
 	return gas.New(c.MultiStore().GetStore(key), c.GasMeter(), store.DefaultGasConfig())
 }
 
+func (c Context) Store(key store.StoreKey) store.Store {
+	return c.MultiStore().GetStore(key)
+}
+
```

Comment:
I can understand the rename to GasStore, but I don't think we should expose a gas-less `Store` method, especially when getting the underlying store is still trivial and exported.

Reply:
```suggestion
func (c Context) GaslessStore(key store.StoreKey) store.Store {
	return c.MultiStore().GetStore(key)
}

```

?

Reply:
> I can understand the rename to GasStore, but I don't think we should expose a gas-less `Store` method, especially when getting the underlying store is still trivial and exported.

The purpose here is not to rename the store but to separate gas store access from non-gas store access in the context. This separation is necessary because we need to wrap the gas meter at different levels for Auth Keeper gas consumption and VM gas consumption.

Please refer to the diagram above. In Auth Keeper, the GasStore wraps the gas meter at the underlying base store level. In VM Keeper, the gas meter is wrapped above the underlying store in the gnoStore, allowing us to capture activity to persist realm objects. The ctx.Store is used by the VM Keeper to access the underlying store without the gas meter, so it can wrap it at a higher layer.

The context.Store abstracts away the underlying MultiStore concept. This approach is consistent with how context, app, and keeper store access are managed.